### PR TITLE
Support generic in json (aka typed json)

### PR DESCRIPTION
**Pull request by sergiodxa**
Right now, if you want to ensure you pass the correct data to `json` when returning a response in a loader/action you need to do something like this:

```ts
type LoaderData = { ... }

export let loader: LoaderFunction = async ({ request }) => {
  // variable created just to ensure the object passed to json match the type
  let data: LoaderData = { ... }
  return json(data);
}

export default function Screen() {
  let data = useLoaderData<LoaderData>()
  // ...
}
```

With this super small change, you don't need that extra variable for type safety anymore

```ts
type LoaderData = { ... }

export let loader: LoaderFunction = async ({ request }) => {
  // no more variables, just pass the generic to json, TS will check if data match the type
  return json<LoaderData>(data);
}

export default function Screen() {
  let data = useLoaderData<LoaderData>()
  // ...
}
```

Super simple change, a little less boilerplate, and if you don't pass the generic it will infer it from the data object.

This makes https://github.com/sergiodxa/remix-utils#typed-json useless.

### CODE REVIEW

File: packages/remix-server-runtime/responses.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * A JSON response. Converts `data` to JSON and sets the `Content-Type` header.
  */
-export function json(data: any, init: number | ResponseInit = {}): Response {
+export function json<Data>(data: Data, init: number | ResponseInit = {}): Response {
```
Comment (MichaelDeBoey) [inline packages/remix-server-runtime/responses.ts:4]: Defaulting `Data` to `unknown` makes sure we don't **have** to pass the type

```suggestion
export function json<Data = unknown>(data: Data, init: number | ResponseInit = {}): Response {
```

**Review by MichaelDeBoey — APPROVED** [review]

File: packages/remix-server-runtime/responses.ts (hunk 1 above)
Comment (sergiodxa) [reply packages/remix-server-runtime/responses.ts:4]: That's not really needed, TS infer it because it's being used in the param (just tried again)

Comment (MichaelDeBoey) [reply packages/remix-server-runtime/responses.ts:4]: Forget what I said then 🙈

**Comment by eps1lon** [comment]
Alternatives: https://github.com/remix-run/remix/issues/97

**Review by ryanflorence — APPROVED** [review]
@mjackson I'd like this, you?

**Pull request merged**